python/dolfinx/plot.py

```diff
@@ -32,39 +33,44 @@ def create_vtk_mesh(mesh: _mesh.Mesh, dim: int, entities=None):
     indices for the associated entry in the mesh geometry.
 
     """
+    cell_type = mesh.topology.cell_type
+    degree = mesh.geometry.cmap.degree
+    if cell_type == _mesh.CellType.prism:
+        raise RuntimeError("Plotting of prism meshes not supported")
+
+    # Use all cells local to process if not supplied
     if entities is None:
         num_cells = mesh.topology.index_map(dim).size_local
         entities = np.arange(num_cells, dtype=np.int32)
     else:
         num_cells = len(entities)
 
-    # Get the indices in the geometry array that correspong to the
-    # topology vertices
-    # NOTE: This linearizes higher order geometries
-    geometry_entities = _cpp.mesh.entities_to_geometry(mesh, dim, entities, False)
+    if dim == mesh.topology.dim:
+        geometry_entities = _cpp.io.extract_vtk_connectivity(mesh)[entities]
```

Comment:
I find this variable name confusing. Something like `vtk_cells` or  `vtk_topology` would be clearer.